include/seqan3/range/view/interleave.hpp

```diff
@@ -0,0 +1,349 @@
+     * \param[in] _step_size The step size for the insertion.
+     * \param[in] _inserted_range The range to be inserted.
+     */
+    view_interleave(urng_t _urange, size_t const _step_size, inserted_rng_t _inserted_range) :
+        urange{_urange}, step_size{_step_size}, inserted_range{_inserted_range}
+    {}
+
+    /*!\brief Construct from a ViewableRange urange and inserted_range by wrapping in a view::all.
+     * \param[in] _urange The underlying range.
+     * \param[in] _step_size The step size for the insertion.
+     * \param[in] _inserted_range The range to be inserted.
+     */
+     template <std::ranges::RandomAccessRange orng_t, std::ranges::RandomAccessRange oirng_t>
+         //!\cond
+         requires std::ranges::SizedRange<orng_t> && std::ranges::SizedRange<oirng_t> &&
+                  std::ranges::ViewableRange<orng_t> &&
+                  std::Same<reference_t<orng_t>, reference_t<oirng_t>>
+    view_interleave(orng_t _urange, size_t const _step_size, oirng_t _inserted_range) :
+        view_interleave{view::all(_urange), _step_size, view::persist(_inserted_range)}
+    {}
+    //!\}
+
+    /*!\name Iterators
+     * \{
+     */
+    /*!\brief Returns an iterator to the first element of the container.
+     * \returns Iterator to the first element.
+     *
+     * If the container is empty, the returned iterator will be equal to end().
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    iterator begin() noexcept
+    {
+        return {*this, 0};
+    }
+
+    //!\copydoc begin()
+    const_iterator begin() const noexcept
+    {
+        return {*this, 0};
+    }
+
+    //!\copydoc begin()
+    const_iterator cbegin() const noexcept
+    {
+        return begin();
+    }
+
+    /*!\brief Returns an iterator to the element following the last element of the container.
+     * \returns Iterator to the first element.
+     *
+     * This element acts as a placeholder; attempting to dereference it results in undefined behaviour.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    iterator end() noexcept
+    {
+        return {*this, size()};
+    }
+
+    //!\copydoc end()
+    const_iterator end() const noexcept
+    {
+        return {*this, size()};
+    }
+
+    //!\copydoc end()
+    const_iterator cend() const noexcept
+    {
+        return end();
+    }
+    //!\}
+
+    /*!\brief Returns the number of elements in the view.
+     * \returns The number of elements in the view.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     */
+    size_type size()
+    {
+        return urange.size() + (std::floor(urange.size() / step_size) * inserted_range.size());
+    }
+
+    //!\copydoc size()
+    size_type size() const
+    {
+        return urange.size() + (std::floor(urange.size() / step_size) * inserted_range.size());
```

Comment:
do not use `.size()` member functions, instead use `std::ranges::size()` free function.

Reply:
also in other places